go.mod

```diff
@@ -6,13 +6,15 @@ require (
 	github.com/apparentlymart/go-cidr v1.0.0
 	github.com/go-logr/logr v1.2.4
 	github.com/google/go-cmp v0.5.9
-	github.com/openshift/api v0.0.0-20231024112103-79b9cd5e6020
+	github.com/openshift/api v0.0.0-20231102215821-c90b4c82852e
```

Comment:
This has to be updated to a much later version.  There have been numerous changes in feature gates and CRD generation that you'll need to pick up.

Reply:
I bumped the `openshift/api` to the latest version in this commit (https://github.com/openshift/cluster-dns-operator/pull/394/commits/ad638deb039868811ac50ab1c066ed0d7dc719b0). I fixed the issues with manifests moved to a different directory by using blank imports in a `manifests/tools.go`. PLMK if this is fine.

Reply:
We have to update controller-runtime, which I've done in https://github.com/openshift/cluster-dns-operator/pull/408. #408 is approved, just waiting for CI to clear.  You can rebase from that for the go.mod changes.

In the *-generated-crd.sh files, don't change the destination file name, just the source file name.

The `tools/tools.go` file should be updated with new imports as @Miciah did in https://github.com/openshift/cluster-ingress-operator/blob/master/tools/tools.go; we shouldn't need `manifest/tools.go`.

The commands you run should include `make generate` as the last step before committing/pushing.

Reply:
#408 is merged into the master. I have rebased this PR. I have moved the blank imports to `tools/tools.go`. Ran the `make generate` and added the same in the commit message.

> In the *-generated-crd.sh files, don't change the destination file name, just the source file name.

I have only changed the destination file names of the DNSNameResolver CRDs to the ones that is there in the openshift/api repo. The names of the CRD files have changed since the last version bump and as we are anyway introducing the files in this PR, I used the same name as the ones in the openshift/api.